src/utils/formHelperFunctions.js

```diff
@@ -385,17 +392,30 @@ function shouldElementHide({ fields, values, element }) {
 function calculateHide(hideIf, fields, values, repeat) {
   // Get the target value (taking repeated elements into account)
   let value = values[hideIf.field_key];
-  if (Array.isArray(value)) value = value[repeat];
+  if (repeat !== undefined && Array.isArray(value)) value = value[repeat];
 
   // If the hideIf value is an empty string, we want to match on the "empty" value of a field
   // This could be null, undefined, an empty array, an empty string, or false
   // Otherwise, just match the hideIf value
   const matchValues =
     hideIf.value === '' ? [null, undefined, [], '', false] : [hideIf.value];
 
-  return hideIf.comparison === 'equal'
-    ? matchValues.includes(value)
-    : !matchValues.includes(value);
+  const matchFn = (val) =>
+    hideIf.comparison === 'equal'
+      ? matchValues.includes(val)
+      : !matchValues.includes(val);
+
+  if (Array.isArray(value)) {
+    return value.some((val) => {
+      if (Array.isArray(val)) {
```

Comment:
can you leave a quick comment explaining when 2D arrays happen?

Reply:
I realized that I don't need to handle 2d array here - because when evaluating a repeated row this function is only exposed to that row's value. So it will just see a normal array, and can use the same logic that button group uses when not repeating